client/src/components/providers/DatasetProvider.js

```diff
@@ -24,4 +25,4 @@ async function getDatasetAttributes({ id }) {
 }
 
 export const DatasetAttributesProvider = SingleQueryProvider(getDatasetAttributes);
-export default SingleQueryProvider(getDataset);
+export default SingleQueryProvider(getDataset, stateIsTerminal);
```

Comment:
This means that a dataset is not updated if e.g. the dataset attributes such as build or datatype change but I still think that this approach is better for now if it helps resolving the underlying problem.

Reply:
Piling up requests is because `setInterval` just fires off in a given interval, that is not why I did this change. We can't poll forever for dataset changes, that is too heavy on the server.

Reply:
A better solution In a history context is that ideally you'd watch for updates in a central polling scheme. When updates come in you'd update a central store with the datasets that have changed, and the single item providers would then automatically get the update. That is what pouchdb did for us previously.